From 000054d63892e10e26316b09d60e7dbe48a25f46 Mon Sep 17 00:00:00 2001
From: Job Noorman <jnoorman@igalia.com>
Date: Wed, 14 Oct 2026 11:04:46 +0200
Subject: [PATCH] tu: fix texel buffer emulation bounds check

We should use an unsigned comparison to check if the offset is small
enough for the size. Otherwise, something like this might write before
the texel's view:

imageStore(texelsOut, -1, vec4(0.5, 0.5, 0.5, 0.5));

Signed-off-by: Job Noorman <jnoorman@igalia.com>
Fixes: dc1bb7bbf45 ("tu: Add option to raise the maximum texel buffer size")
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/45204>
---
 src/freedreno/vulkan/tu_shader.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/freedreno/vulkan/tu_shader.cc b/src/freedreno/vulkan/tu_shader.cc
index 45e17323..0c79d5db 100644
--- a/src/freedreno/vulkan/tu_shader.cc
+++ b/src/freedreno/vulkan/tu_shader.cc
@@ -632,7 +632,7 @@ build_texel_buffer_as_image_coords(nir_builder *b,
 {
    nir_def *base_offset = nullptr;
    nir_def *real_size = build_texel_buffer_size(b, desc, &base_offset);
-   nir_def *oob = nir_ige(b, offset, real_size);
+   nir_def *oob = nir_uge(b, offset, real_size);
 
    offset = nir_iadd(b, offset, base_offset);
 
